torch/utils/_config_module.py

```diff
@@ -153,69 +154,84 @@ def __getattr__(self, name: str) -> Any:
             raise AttributeError(f"{self.__name__}.{name} does not exist") from e
 
     def __delattr__(self, name: str) -> None:
+        self._is_dirty = True
         # must support delete because unittest.mock.patch deletes
         # then recreate things
         del self._config[name]
 
+    def _get_dict(
+        self,
+        ignored_keys: Optional[List[str]] = None,
+        ignored_prefixes: Optional[List[str]] = None,
+        skip_default: bool = False,
+    ) -> Dict[str, Any]:
+        config: Dict[str, Any] = {}
+        for key in self._config:
+            if ignored_keys and key in ignored_keys:
+                if skip_default and self._config[key] != self._default[key]:
+                    warnings.warn(
+                        f"Skipping serialization of {key} value {self._config[key]}"
+                    )
+                continue
+            if ignored_prefixes:
+                if any(key.startswith(prefix) for prefix in ignored_prefixes):
+                    continue
+            if skip_default and self._config[key] == self._default[key]:
+                continue
+            config[key] = copy.deepcopy(self._config[key])
+        return config
+
     def save_config(self) -> bytes:
         """Convert config to a pickled blob"""
-        config = dict(self._config)
-        for key in config.get("_save_config_ignore", ()):
-            config.pop(key)
-        return pickle.dumps(config, protocol=2)
+        return pickle.dumps(
+            self._get_dict(ignored_keys=self._config.get("_save_config_ignore", ())),
```

Comment:
It occurs to me that now that we're supporting field() style definitions of config values, we don't have to defines these out of band anymore, we could just have them be magic objects that are processed on load and store them separately. 🤔

Reply:
i.e. foo = Config(value=True, skip_save_config=True)?

Happy to send a PR :).